tedana/reporting/static_figures.py

```diff
@@ -574,7 +574,11 @@ def plot_t2star_and_s0(
     s0_img = io_generator.get_name("s0 img")
     mask_img = io.new_nii_like(io_generator.reference_img, mask.astype(int))
     assert os.path.isfile(t2star_img), f"File {t2star_img} does not exist"
-    assert os.path.isfile(s0_img), f"File {s0_img} does not exist"
+
+    # Check if S0 image exists, issue warning if not
+    s0_exists = os.path.isfile(s0_img)
+    if not s0_exists:
+        warnings.warn(f"File {s0_img} does not exist. S0 plots will not be generated.")
```

Comment:
This should be an `LGR.warning` not `warnings.warn` so that it's stored in the logger. I'd actually prefer that this is an `LGR.info` since this is just describing expected functionality and isn't actually a problem. That is "S0 maps and T2* fit metrics are not in report since a pre-existing T2* map was provided"